tests/e2e/test_e2e_workflows.py

```diff
@@ -0,0 +1,399 @@
+                    Comment(
+                        id=UUID("550e8400-e29b-41d4-a716-446655440102"),
+                        ticket_id=bug_id,
+                        author="developer@example.com",
+                        content="Fixed CSS media query issue in PR #456",
+                        created_at=datetime.fromisoformat("2024-01-21T14:00:00+00:00"),
+                    ),
+                ],
+            ),
+        }
+
+        client = TestClient(app)
+
+        # User reports bug
+        mock_jira_backend["create_ticket"].return_value = states["reported"]
+        response = client.post(
+            "/api/v1/tickets",
+            json={
+                "title": "Login button not visible on mobile",
+                "description": "The login button disappears on mobile devices",
+                "reporter": "user@example.com",
+                "priority": "medium",
+            },
+            headers={"X-User-ID": "test-user", "X-Project-Key": "BUG"},
+        )
+        assert response.status_code == HTTP_CREATED
+
+        # Triager increases priority and assigns
+        mock_jira_backend["update_ticket"].return_value = states["triaged"]
+        response = client.patch(
+            f"/api/v1/tickets/{bug_id}",
+            json={
+                "priority": "critical",
+                "assignee": "developer@example.com",
+            },
+            headers={"X-User-ID": "test-triager", "X-Project-Key": "BUG"},
+        )
+        assert response.status_code == HTTP_OK
+
+        # Developer starts work
+        mock_jira_backend["update_ticket"].return_value = states["in_progress"]
+        response = client.patch(
+            f"/api/v1/tickets/{bug_id}",
+            json={"status": "in_progress"},
+            headers={
+                "X-User-ID": "test-developer",
+                "X-Project-Key": "BUG",
+            },
+        )
+        assert response.status_code == HTTP_OK
+
+        # Developer fixes and adds comment
+        mock_jira_backend["add_comment"].return_value = states["resolved"].comments[0]
+        response = client.post(
+            f"/api/v1/tickets/{bug_id}/comments",
+            json={"author": "developer@example.com", "content": "Fixed CSS media query issue in PR #456"},
+            headers={
+                "X-User-ID": "test-developer",
+                "X-Project-Key": "BUG",
+            },
+        )
+        assert response.status_code == HTTP_CREATED
+
+        # Mark as resolved
+        mock_jira_backend["update_ticket"].return_value = states["resolved"]
+        response = client.patch(
+            f"/api/v1/tickets/{bug_id}",
+            json={"status": "resolved"},
+            headers={
+                "X-User-ID": "test-developer",
+                "X-Project-Key": "BUG",
+            },
+        )
+        assert response.status_code == HTTP_OK
+
+
+class TestE2EDataConsistency:
+    """E2E tests for data consistency across operations."""
```

Comment:
Does it matter to test whether the data is consistent? It's handled by Jira, not by your code... (Besides in a real-world scenario your client is not the only one making changes, making the test moot.)